dipy/reconst/tests/test_ivim.py

```diff
@@ -65,7 +65,7 @@
                         500., 600., 700., 800., 900., 1000.])
 
 bvecs_no_b0 = generate_bvecs(N)
-gtab_no_b0 = gradient_table(bvals_no_b0, bvecs.T)
+gtab_no_b0 = gradient_table(bvals_no_b0, bvecs.T, b0_threshold=0)
```

Comment:
The problem here is that it would break backwards compatibility for people who are already using IVIM with a gradient table initialized with the default parameters. I think that we would need to check and provide a better error for people who are using their code as it was.

Reply:
@arokem Thank you for this! Makes sense to me.. Should I set up a warning?

Reply:
I think this PR is almost ready to go. the only point is I have no clue to solve this backward compatibility problem.  I do not think a warning message will solve this problem and we have already [this check](https://github.com/nipy/dipy/blob/master/dipy/reconst/ivim.py#L216). 

Any suggestion?

Reply:
I think that we need to do this within the IVIM implementation. One way to deal with this would be to check whether the gtab provided has a non-zero b0_threshold, and raise an error when that's the case, unless the user gives some other indication that that's intended. It's a bit kludgy, I'll admit it. Open to other suggestions.

Reply:
@arokem Will setting the default value of `b0_threshold` to `50` have repercussions in the functioning of IVIM for current users? I'm not sure I understand. Please could you help me with  understanding this?

Reply:
The problem is that for IVIM, a user might very reasonably have a measurement with a b-value of 40, that should not be considered a b0 measurement. In previous versions of the software, when they ran the default code, it wasn't. But with this change introduced that measurement will now be considered to have a b-value of 0, and they wouldn't have any way of knowing that this change was introduced.